jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/RetainableByteBuffer.java

```diff
@@ -159,6 +170,419 @@ default void clear()
         BufferUtil.clear(getByteBuffer());
     }
 
+    default int append(byte[] bytes, int offset, int length)
```

Comment:
I'd rather not have variants of `append` working on `byte[]` unless necessary, and so far, only tests use this variant.

Reply:
I think it is worth while if we ever want this API to replace much of BufferUtil.  In fact I could probably find usages for this now by replace BufferUtil calls.